Flexbox

cmigz opened the pull request:
Hope you like it!  Let me know if you notice anything that needs to be fixed =) 

Did a brief tutorial on flexbox, want to expand more on it but has the basics for now.

cmigz committed:
Create Directory / File

flexbox/flex-basics.md:
- Made directory to store files on topic
- Wrote basic starter structure for md
  - Create
    - Header with topic
    - Start of table of contents
    - Introduction with content and quote from MDN

cmigz committed:
Create Section: The Basics

flexbox/flex-basics.md:
- Create sections for basics on flex box
- Add two new display values with visual examples and css examples

cmigz committed:
Create Manipulating Contents Section / Gram Fix

flexbox/flex-basics.md:
- Fix grammar on Display: inline-flex section
- Create Section Manipulating Contents of Flexbox Container
  - Fill with ol of methods and sub categories
  - Fill content of Order & Direction with write ups/examples
  - Start section on Alignment, got as far as Flex-Start

cmigz committed:
Finish remaining content for Alignment

flexbox/flex-basics.md:
- Create sections flex-end, center, space-between, and space-around
  - Filled each with related snippets and visual examples

cmigz committed:
Start expansion on manipulation sec points

flexbox/flex-basics.md:
- Playing with new idea to expand on section
- Define default value, possible values per attribute, and a brief
  write up on what the attribute's purpose is.

cmigz committed:
Finish Content

- Fill remaining content
- Create rest of table of content
- Create Sources and External Resources
- Ran all text through grammarly

laurpaik-zz commented on flexbox/flex-basics.md:
````diff
@@ -0,0 +1,245 @@
+    display: inline-flex;
+  }
+  ```
+  ![alt text](https://i.imgur.com/NtlTQyC.png "display: inline-flex;")
````
This section seems a bit long visually since the images are kind of large. Would it be easier to just put a gif or something?
This is the gif I made [testing this out](https://codepen.io/laurpaik/pen/bYEyZg?editors=1100)-- Had a hard time figuring out `inline-flex`, so my `inline-flex` example isn't very good.
![](https://cl.ly/1O3F0t2B062K/Screen%20Recording%202017-11-03%20at%2010.48%20AM.gif)

laurpaik-zz commented on flexbox/flex-basics.md:
```diff
@@ -0,0 +1,245 @@
+
+Ok! So FlexBox!
+
+Flexbox or "Flexible Box" is a relatively new set of CSS properties we can utilize to distribute space among elements and align content exactly how we want.  It's never been easier to set up grids and organize content, I assure you once you get the swing of Flexbox you will never bother with float or any other grid system again.
```
This can probably be broken down into two sentences :)
```diff
- It's never been easier to set up grids and organize content, I assure you once you get the swing of Flexbox you will never bother with float or any other grid system again.
+ It's never been easier to set up grids and organize content. I assure you once you get the swing of Flexbox you will never bother with `float` or any other grid system again.
```

laurpaik-zz commented on flexbox/flex-basics.md:
````diff
@@ -0,0 +1,245 @@
+  ```
+  ![alt text](https://i.imgur.com/NtlTQyC.png "display: inline-flex;")
+
+  These display properties create flex containers which make a "flex formatting context" for its content.  Similar to making a block formatting context but the flex layout is the replacement with flexbox's built-in ability to adjust automatically.
````
I had to read that second sentence a few times. Is there a better way to rephrase it for accessibility's sake?
Suggestion, if my meaning is the same as yours:
```diff
-Similar to making a block formatting context but the flex layout is the replacement with flexbox's built-in ability to adjust automatically.
+ It's similar to making a block formatting context, but here we use flex layout for that built-in ability to adjust automatically.
```

laurpaik-zz commented on flexbox/flex-basics.md:
```diff
@@ -0,0 +1,245 @@
+
+  These display properties create flex containers which make a "flex formatting context" for its content.  Similar to making a block formatting context but the flex layout is the replacement with flexbox's built-in ability to adjust automatically.
+
+## Manipulating Contents of Flexbox Container
```
Love that you include the default values here! Could you please use tic-marks for all the keywords in the values, defaults, and descriptions?
example:
```diff
-    * **flex-flow:** row nowrap || column-reverse || column wrap || row-reverse wrap-reverse;
+   * **flex-flow:** `row nowrap` || `column-reverse` || `column wrap` || `row-reverse wrap-reverse`;
-      * _Default:_ row nowrap
+      * _Default:_ `row nowrap`
-      * Since flex-direction and flex-wrap are often used in tandem we have the flex-flow property to use as a shortcut for setting the two in one property.
+      * Since `flex-direction` and `flex-wrap` are often used in tandem we have the `flex-flow` property to use as a shortcut for setting the two in one property.
```

laurpaik-zz commented on flexbox/flex-basics.md:
```diff
@@ -0,0 +1,245 @@
+
+### Justify Content
+
+This property allows for the alignment of children along the primary axis of the flex container.  So if we have a parent container with row we will be justifying content along the X-Axis while if it were a column we'd be justifying it along the Y-Axis.
```
```diff
-So if we have a parent container with row we will be justifying content along the X-Axis while if it were a column we'd be justifying it along the Y-Axis.
+ So if we have a parent container with `flex-direction: row` we will be justifying content along the X-Axis while with `flex-direction: column` we'd be justifying it along the Y-Axis.
```

laurpaik-zz commented:
Hey @cmigz ! I want to introduce myself before looking over this, since it's your first time contributing (and since Ben seems to be a bit busy right now)! My name is Lauren, and I was in Boston's WDI-016. Let me know if you have any questions on the feedback when I send it!

laurpaik-zz commented on flexbox/flex-basics.md:
```diff
@@ -0,0 +1,245 @@
+
+A lot of this walkthrough has come from three sources I used to pick up and practice the basics of flex-box.  You will find much more in-depth information from these links and the chance to practice and play with all the different properties.  Please get in touch if you find any additional resources you think would make a good addition to this repository!
+
+- [CodeSchool: "Craking the Case with Flexbox"](https://www.codeschool.com/courses/cracking-the-case-with-flexbox)
```
```diff
- "Craking the Case with Flexbox"
+ "Cracking the Case with Flexbox"
```

laurpaik-zz commented on flexbox/flex-basics.md:
```diff
@@ -0,0 +1,245 @@
+A lot of this walkthrough has come from three sources I used to pick up and practice the basics of flex-box.  You will find much more in-depth information from these links and the chance to practice and play with all the different properties.  Please get in touch if you find any additional resources you think would make a good addition to this repository!
+
+- [CodeSchool: "Craking the Case with Flexbox"](https://www.codeschool.com/courses/cracking-the-case-with-flexbox)
+  - Comes with the silly songs that go along with all of code school's content.  All joking aside this is the most in-depth and best source I found for really getting into the details with flexbox but more importantly the chance to practice use the properties and see them in real time.
```
```diff
- all of code school's content.
+ all of CodeSchool's content.
```
Last sentence is crazy long! Lol, suggestion:
```diff
- All joking aside this is the most in-depth and best source I found for really getting into the details with flexbox but more importantly the chance to practice use the properties and see them in real time.
+ All jokes aside, this is the most in-depth source I found for getting into the details of Flexbox. It gives you a chance to practice using the properties and see the changes in real time.
```

cmigz commented:
Hi @laurpaik! It's nice to speak with you, my name is Christian.  Looking forward to your feedback, thanks for looking it over!

laurpaik-zz commented on flexbox/flex-basics.md:
```diff
@@ -0,0 +1,245 @@
+- [Flexbox Froggy](http://flexboxfroggy.com/)
+  - A great little game/practice site.  It has 24 levels that teach/test use of each of flexbox's properties.  Really good for solidifying what you've learned.
+- [Flexbox Cheat Sheet](https://yoksel.github.io/flex-cheatsheet/)
+  - My go-to source when I forget what it is specifically I'm trying to accomplish.  It's really useful in that it has a section for each property and interactive example where you can select a property value and see how it affects the container and it's children.
```
Also really long sentence!
Suggestion:
```diff
- It's really useful in that it has a section for each property and interactive example where you can select a property value and see how it affects the container and it's children.
+ It has a section for each property with interactive examples.
```

laurpaik-zz commented on flexbox/flex-basics.md:
````diff
@@ -0,0 +1,245 @@
+```
+![alt text](https://i.imgur.com/Gzc2uEn.png "justify-content: space-around;")
+
+## Sources and External Resources
````
Another great resource, if you want to add it: WesBos has a free, 20-minute tutorial called [What The Flexbox?!](https://flexbox.io/). His courses are kind of code-along style, with example files you can fork and clone on GitHub.
[This](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) is another great resource too :)

cmigz commented:
Haha  @laurpaik  I'm so glad to have had another set of eyes go over this, I never realize in the moment how much I tend to write run-on verbose sentences lol.  Also love the suggestions for the tic marks for the property values and such, great idea.  Soon as I get the chance I'll make all the suggested changes/additions and submit a new PR.  I appreciate the fine-tooth combing, love this kind of collaboration =)

laurpaik-zz commented:
@cmigz you actually won't need to submit a new PR if you push changes to the same branch. The new commits will simply be added to this PR, which'll be nice because I can see my suggestions before your changes too and we'll both get a sort of history log 😄 If you want an example, you can check out some of the closed pull requests that have a bunch of comments on them, like [this](https://github.com/BenGitsCode/today-i-learned/pull/29).

cmigz committed:
Fixes Run On Sentence, Replaces Static Ex w/ Gif

flex-basics.md =>
- Splits run on in intro paragraph into two sentences
- Replaces display: flex, and display:inline-flex w/ gif examples

cmigz committed:
Adds tics to keywords for values, defaults, descr

flex-basics.md =>
- Surrounds all keyword values in tic marks
- Fixes run on sentence in display inline-flex paragraph

cmigz committed:
Updates Example Images

flex-basics.md =>
- Updates all example images with smaller examples and gifs where useful

cmigz committed:
Adds More Tics, Updates Title, Adds Resource

flex-basics.md =>
- Fixes a few more run on sentences
- Adds thanks to Lauren for her suggestions
- Adds suggested resource from Lauren's noes
- Adds tics where they were missed

cmigz commented:
Hey @laurpaik!

Sorry I didn't do your fixes sooner!  Hope you had a great holiday and new year! 

Hope I hit all your suggestions, feel free to let me know anything I missed or any new edits when you get a chance.
